llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp

```diff
@@ -178,6 +178,20 @@ bool AMDGPUAtomicOptimizerImpl::run(Function &F) {
   return Changed;
 }
 
+static bool shouldOptimize(Type *Ty) {
```

Comment:
Better name that expresses why this type is handleable. 

Also in a follow up, really should cover the i16/half/bfloat and 2 x half, 2 x bfloat cases